src/index.js

```diff
@@ -307,7 +307,16 @@ function cloneElement(element, props, ...children) {
 		elementProps,
 		element.children || elementProps && elementProps.children
 	);
-	return normalizeVNode(preactCloneElement(node, props, ...children));
+	// Only provide the 3rd argument if needed.
+	// Arguments 3+ overwrite element.children in preactCloneElement
+	let cloneArgs = [node, props];
+	if (children && children.length) {
```

Comment:
I actually wonder (was going to modify and run the tests) if the block on L313 can get dropped?  `preactCloneElement()` will look for `props.children` if its arity is <=2 and it seems like we're passing it `props.children` in that case.